### PR TITLE
Add global Molmer-Sorenson gate

### PR DESCRIPTION
**Pull request by ajavadia**
Molmer-Sorenson gate on a variable number of qubits. 

Decomposes to a bunch of 2-qubit XX interactions.

This also updates the ion-trap branch to sync with master.

### CODE REVIEW

File: qiskit/extensions/standard/ms.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        rule = []
+        for i in range(self.num_qubits):
+            for j in range(i+1, self.num_qubits):
+                rule += [(RXXGate(self.params[0]), [q[i], q[j]], [])]
+
+        for inst in rule:
+            definition.append(inst)
+        self.definition = definition
+
+        print(self.num_qubits)
```
Comment (ewinston) [inline qiskit/extensions/standard/ms.py]: don't need these anymore

File: qiskit/extensions/standard/ms.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        print(self.definition)
+
+
+
+    #def to_matrix(self):
```
Comment (ewinston) [inline qiskit/extensions/standard/ms.py]: drop

**Comment by ajavadia** [comment]
rebase issues are solved now.

File: qiskit/extensions/standard/ms.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        for i in range(self.num_qubits):
+            for j in range(i+1, self.num_qubits):
+                rule += [(RXXGate(self.params[0]), [q[i], q[j]], [])]
+
+        for inst in rule:
+            definition.append(inst)
+        self.definition = definition
+
+
+def ms(self, theta,  qubits):
```
Comment (kdk) [inline qiskit/extensions/standard/ms.py]:
```suggestion
def ms(self, theta, qubits):
```

File: qiskit/extensions/standard/ms.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name
+
+"""
+Molmer-Sorensen gate.
+"""
+
+import numpy
```
Comment (kdk) [inline qiskit/extensions/standard/ms.py]:
```suggestion
```

**Pull request merged**